src/globus_sdk/services/compute/client.py

```diff
@@ -12,13 +12,14 @@
 log = logging.getLogger(__name__)
 
 
-class ComputeClient(client.BaseClient):
+class ComputeClientV2(client.BaseClient):
     r"""
-    Client for the Globus Compute API.
+    Client for the Globus Compute API, version 2.
 
-    .. automethodlist:: globus_sdk.ComputeClient
+    .. automethodlist:: globus_sdk.ComputeClientV2
     """
 
+    base_path = "/v2/"
```

Comment:
I'm not certain, but I don't think we want to set `base_path` for any clients. It's a client attribute that I would like to deprecate and remove at some point, though I don't know if we ever will (since it would cause pretty wide breakage to do so, to the point that I'm not sure it's worth it even in a major release).

The issue I have with it is that it makes it impossible to call unversioned routes from outside of this path prefix using a client.

That is, it changes the behavior of `ComputeClient(...).get("/_foo")` to hit `/v2/_foo`. Which means that if you really _want_ to issue a `GET /v2/_foo`, you need to...
```python
class MyComputeClient(ComputeClientV2):
    base_path = "/"

MyComputeClient(...).get("/_foo")
```

In any case, if we _do_ set the base_path here, then the `ComputeClient` class needs to reset it to `"/"` in order to remain backwards compatible.

Reply:
The `GroupsClient` and `TransferClient` both set this attribute, so figured it was fine, but I'm happy to remove it.

Reply:
I think we should remove it. I'll add a comment on those two usages to indicate that we may want to remove them in a future release, so that it's clearer.
Note that this is primarily my own opinion -- it's possible that my comment PR won't be accepted by the team, but at least it will give us some clarity.